src/common/layers/json.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::BTreeMap;
+
+/// A layer for the tracing framework to record structured event and span data.
+#[derive(Debug)]
+pub struct HyperLayer;
```

Comment:
i'm not sure if i understand the motivation for `hyper` providing its own `tracing_subscriber::Layer` implementations? the core idea of `tracing` is that it serves as a _facade_ for connecting instrumentation in libraries with subscribers defined in _other_ libraries. right now, it looks like this `Layer` doesn't do any kind of `hyper`-specific behavior...so why is it part of `hyper` rather than a separate crate providing a general-purpose JSON layer?

if the intention is to eventually add hyper-specific behavior, i think a better approach is still probably to include that layer in a separate crate --- call it `hyper-tracing`, or something? that way, the application can plug in whatever `tracing` layer it needs, and `hyper`'s core functionality --- actually implementing HTTP clients and servers --- just exposes `tracing` instrumentation.